pkg/dhcp6client/packetSock.go

```diff
@@ -54,6 +55,12 @@ func (pc packetSock) Close() error {
 	return syscall.Close(pc.fd)
 }
 
+// Set a read timeout
```

Comment:
SetReadTimeout functions for v4 and v6 are the same. However those functions are member function of packetSocks in v4 and v6, and the packSocks are different. Even though we are able to implement only one function to set timeout for both v4 and v6 packetSocks, we still have to write a new of our own, instead of just using the one that was written by d2g.

Hence I might prefer to leave them separate.